airbyte-integrations/connectors/source-datascope/source_datascope/source.py

```diff
@@ -0,0 +1,160 @@
+        if str(self.schema_type) == 'dynamic':
+            return self._request(f"{self.BASE_URL}/api/external/v2/airbyte_schema?token={self.token}&form_id={self.form_id}")[0]
+        else:
+            return schema
+
+    def parse_response(
+            self,
+            response: requests.Response,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        return response.json()
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+        # This method is called once for each record returned from the API to compare the cursor field value in that record with the current state
+        # we then return an updated state object. If this is the first time we run a sync or no state was passed, current_stream_state will be None.
+        if current_stream_state is not None and 'updated_at' in current_stream_state:
+            current_parsed_date = datetime.strptime(current_stream_state['updated_at'], '%Y-%m-%dT%H:%M:%S%z')
+            last_date = latest_record['updated_at'].split('.')[0] + '+0000'
+            if self.schema_type == 'dynamic':
+                latest_record_date = datetime.strptime(last_date, '%d/%m/%Y %H:%M%z')
+            else:
+                latest_record_date = datetime.strptime(last_date, '%Y-%m-%dT%H:%M:%S%z')
+            return {'updated_at': max(current_parsed_date, latest_record_date).strftime('%Y-%m-%dT%H:%M:%S%z')}
+        else:
+            return {'updated_at': self.start_date.strftime('%Y-%m-%dT%H:%M:%S%z')}
+
+    def _chunk_date_range(self, start_date: datetime) -> List[Mapping[str, any]]:
+        """
+        Returns a list of each day between the start date and now.
+        The return value is a list of dicts {'date': date_string}.
+        """
+        dates = []
+        while start_date < datetime.now(start_date.tzinfo):
+            dates.append({'updated_at': start_date.strftime('%Y-%m-%dT%H:%M:%S%z')})
+            start_date += timedelta(days=7)
+        return dates
+    def stream_slices(self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None) -> Iterable[
+        Optional[Mapping[str, any]]]:
+        start_date = datetime.strptime(stream_state['updated_at'], '%Y-%m-%dT%H:%M:%S%z') if stream_state and 'updated_at' in stream_state else self.start_date
+        return self._chunk_date_range(start_date)
+
+
+
+
+class Forms(DatascopeIncrementalStream):
+    primary_key = "form_answer_id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        if self.schema_type == 'dynamic':
+            return "v2/airbyte"
```

Comment:
I don't see this endpoint in the docs? https://dscope.github.io/

Reply:
It's a specific endpoint to retrieve data only to Airbyte integration

Reply:
Ah I see! Are you the maintainer of the datascope API then? If so, will this be a stable endpoint on the API available to all users? Just trying to make sure we're building on solid ground here.

Reply:
Yes, it will be stable but probably not documented on the API Doc https://dscope.github.io/docs/ because is only for Airbyte use. Do you think We need to change that  and use a public endpoint or is ok?